sourcegraph/sourcegraph
# Add newcomers onboarding tour

Closes https://github.com/sourcegraph/sourcegraph/issues/27177 and https://github.com/sourcegraph/sourcegraph/issues/29420.

## Description
This PR:
- Adds a new onboarding tour for not signed-in users.
- Adds the ability to force override feature flags through initial URL parameters

## How to test
1. `sg start web-standalone`
2. Open the home page and force enable "getting started" tour: 
   1. `https://sourcegraph.test:3443/search?feature-flag-key=getting-started-tour&feature-flag-value=true`
   2. Go through the new tour widget and check it works properly
3. Open the home page and disable the "getting started" tour:
   1. `https://sourcegraph.test:3443/search?feature-flag-key=getting-started-tour`
   2. Check that onboarding tour doesn't show up and everything works as previously

## Screenshots
<details>

| Before | After |
| --: | :-- |
| ![image](https://user-images.githubusercontent.com/6717049/146940073-aeed41f4-dbfb-4186-a6ee-8f3f3643c477.png) | When tour is OPEN ![image](https://user-images.githubusercontent.com/6717049/146940108-71ae3a8e-dcbb-4729-b037-2f7dcea259fa.png) |
| | When tour is CLOSED ![image](https://user-images.githubusercontent.com/6717049/146940843-e77ce825-3407-4ab0-be0b-7d6c3ebc75f8.png) |
| ![image](https://user-images.githubusercontent.com/6717049/146940623-cb6f6d55-b976-4101-9321-6914a1cc0bce.png) | ![image](https://user-images.githubusercontent.com/6717049/146940605-7b15fffb-4c46-4111-918a-ae0fe6ca5289.png) |
| ![image](https://user-images.githubusercontent.com/6717049/146940735-0eaaecf7-3117-4cf3-b2cc-2db6c1b7af3e.png) | ![image](https://user-images.githubusercontent.com/6717049/146940696-3636c2f6-403b-4ef7-9c0c-ff7ef12b51ef.png) |

</details>
